lib/libc/newlib/libc-hooks.c

```diff
@@ -157,3 +162,51 @@ void *_sbrk(int count)
 	}
 }
 FUNC_ALIAS(_sbrk, sbrk, void *);
+
+int _gettimeofday(struct timeval *tv, void *tz)
+{
+#if CONFIG_RTC
+	struct device *rtc_dev = NULL;
+
+	rtc_dev = device_get_binding(CONFIG_RTC_0_NAME);
+	if (!rtc_dev) {
+		errno = ENXIO;
+		return -1;
+	}
+
+	if (tv) {
+		tv->tv_sec = rtc_read(rtc_dev);
+		tv->tv_usec = 0;
```

Comment:
And this related bit is **really** going to annoy people.  The whole purpose to gettimeofday() in Unix (vs. the classic time() system call) is that it supports very fine time units.  Bailing and returning an integer count of seconds is not going to meet the expectations of lots and lots of real world code.